ts/features/idpay/onboarding/xstate/machine.ts

```diff
@@ -269,24 +308,85 @@ const createIDPayOnboardingMachine = () =>
             ]
           }
         },
+
         DISPLAYING_REQUIRED_SELF_CRITERIA: {
           tags: [WAITING_USER_INPUT_TAG],
-          entry: "navigateToSelfDeclarationsScreen",
-          on: {
-            ACCEPT_REQUIRED_SELF_CRITERIA: {
-              target: "ACCEPTING_REQUIRED_CRITERIA"
+          initial: "EVALUATING_SELF_CRITERIA",
+          onDone: {
+            target: "ACCEPTING_REQUIRED_CRITERIA"
+            // triggered by the 'final' substate
+          },
+          states: {
+            EVALUATING_SELF_CRITERIA: {
+              tags: [LOADING_TAG],
+              always: [
+                {
+                  target: "DISPLAYING_BOOL_CRITERIA",
+                  cond: "hasBoolRequiredCriteria"
+                },
+                {
+                  target: "CYCLING_MULTI_CRITERIA",
+                  cond: "hasMultiRequiredCriteria"
+                }
+              ]
             },
-            GO_BACK: [
-              {
-                target: "DISPLAYING_REQUIRED_PDND_CRITERIA",
-                cond: "hasPDNDRequiredCriteria"
-              },
-              {
-                target: "DISPLAYING_INITIATIVE"
+            DISPLAYING_BOOL_CRITERIA: {
+              tags: [WAITING_USER_INPUT_TAG],
+              entry: "navigateToBoolSelfDeclarationsScreen",
+              on: {
+                GO_BACK: [
+                  {
+                    target:
+                      "#IDPAY_ONBOARDING.DISPLAYING_REQUIRED_PDND_CRITERIA",
+                    cond: "hasPDNDRequiredCriteria"
+                  },
+                  {
+                    target: "#IDPAY_ONBOARDING.DISPLAYING_INITIATIVE"
+                  }
+                ],
+                ACCEPT_REQUIRED_BOOL_CRITERIA: [
+                  {
+                    target: "CYCLING_MULTI_CRITERIA",
+                    cond: "hasMultiRequiredCriteria"
+                  },
+                  {
+                    target: "MULTI_PREREQUISITES_CYCLE_OVER"
+                  }
+                ]
               }
-            ]
+            },
+            CYCLING_MULTI_CRITERIA: {
```

Comment:
```suggestion
            DISPLAYING_MULTI_CRITERIA: {
```

Reply:
I believe `CYCLING` is more eloquent than `DISPLAYING`, since while in the state the user is cycling through all the criteria, maybe `DISPLAING_MULTI_CRITERIA_CYCLE`, in order to follow the current `DISPLAYING`-`EVALUATING`-`LOADING` naming convention?

Reply:
`CYCLE` is something that repeats itself, in this case criteria are displayed sequentially and eventually they reach an end. 
Also, `DISPLAYING` it's a word already used to describe other "display" states, so we can have more consistency.